tests/API/CategoriesApiTest.php

```diff
@@ -58,13 +63,55 @@ public function testCategoryCreate(): void
         $this->response->assertOk();
     }
 
-    public function testCategoryDestroy(): void
+    public function testCategoryDestroyUserAdmin(): void
     {
-        $user = User::factory(['email' => 'category@email.com'])->make();
+        $user = config('auth.providers.users.model')::factory()->create();
+        $user->assignRole('admin');
+
         Passport::actingAs($user, ['categories']);
 
         $category = Category::factory()->create();
         $this->response = $this->json('DELETE', '/api/categories/' . $category->getKey());
         $this->response->assertOk();
     }
+
+    public function testCategoryUpdateUserStudent(): void
+    {
+        $user = config('auth.providers.users.model')::factory()->create();
+        $user->assignRole('student');
+        Passport::actingAs($user, ['categories']);
+
+        $category = Category::factory()->create();
+        $this->response = $this->startSession()->json('PUT', '/api/categories/' . $category->getKey(), [
```

Comment:
Można by dać po prostu `$this->actingAs($user, 'api')->json`

Reply:
Zmienione